docs/rule/no-obscure-array-access.md

```diff
@@ -0,0 +1,39 @@
+# no-obscure-array-access
+
+Using obscure expressions `{{list.[1].name}}` is discouraged and is likely to be deprecated soon.
+
+This rule recommends the use of Ember's `get` helper as an alternative for accessing array values.
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+{{foo bar=list.[0]}}
+```
+
+```hbs
+{{foo bar @list.[1]}}
```

Comment:
Is this missing the equals sign?

And I actually think one good and one bad example are probably enough here. We don't need to show every possible variation of the surrounding syntax. Only the specific array access is what matters. And having too many examples can be distracting.

bad

```hbs
<Foo @bar={{list.[0]}} />
```
good

```hbs
<Foo @bar={{get list '0'}} />
```

Reply:
Yeah good catch.

Reply:
Do you think we should remove unnecessary examples as I mentioned?

Reply:
Thoughts on this?

Reply:
I missed that originally, I will do it now.